src/it/java/rest/demo/ToDoControllerIntegrationTest.java

```diff
@@ -0,0 +1,158 @@
+                .andExpect(jsonPath("$.[0].id").value(1))
+                .andExpect(jsonPath("$.[0].name").value("List 1"))
+                .andExpect(jsonPath("$.[0].description").value("List 1 description"))
+                .andExpect(jsonPath("$.[0].tasks").isEmpty())
+                .andExpect(jsonPath("$.[0].createdAt").value(containsString(date.format(formatter))))
+                .andExpect(jsonPath("$.[0].updatedAt").value(containsString(date.format(formatter))))
+                .andExpect(jsonPath("$.[1].id").value(2))
+                .andExpect(jsonPath("$.[1].name").value("List 2"))
+                .andExpect(jsonPath("$.[1].description").value("List 2 description"))
+                .andExpect(jsonPath("$.[1].tasks").isEmpty())
+                .andExpect(jsonPath("$.[1].createdAt").value(containsString(date.format(formatter))))
+                .andExpect(jsonPath("$.[1].updatedAt").value(containsString(date.format(formatter))));
+
+         }
+
+    @Test
+    @DisplayName("When GET and ToDos with Tasks, Then ToDos with Tasks")
+    void whenToDosHaveTasks_thenSuccessfulAndReturnsToDosWithTasks() throws Exception {
+
+        createToDos();
+        createTasks();
+        toDo1.setTasks(tasks);
+        toDoRepository.saveAll(toDos);
+
+        mockMvc.perform(get("/api/todos")
+                .contentType("application/json"))
+                .andExpect(status().is2xxSuccessful())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+                .andExpect(jsonPath("$.[0].id").value(1))
+                .andExpect(jsonPath("$.[0].name").value("List 1"))
+                .andExpect(jsonPath("$.[0].description").value("List 1 description"))
+                .andExpect(jsonPath("$.[0].tasks.[0].id").value(1))
+                .andExpect(jsonPath("$.[0].tasks.[0].name").value("Task 1"))
+                .andExpect(jsonPath("$.[0].tasks.[1].id").value(2))
+                .andExpect(jsonPath("$.[0].tasks.[1].name").value("Task 2"))
+                .andExpect(jsonPath("$.[0].createdAt").value(containsString(date.format(formatter))))
+                .andExpect(jsonPath("$.[0].updatedAt").value(containsString(date.format(formatter))))
+                .andExpect(jsonPath("$.[1].id").value(2))
+                .andExpect(jsonPath("$.[1].name").value("List 2"))
+                .andExpect(jsonPath("$.[1].description").value("List 2 description"))
+                .andExpect(jsonPath("$.[1].tasks").isEmpty())
+                .andExpect(jsonPath("$.[1].createdAt").value(containsString(date.format(formatter))))
+                .andExpect(jsonPath("$.[1].updatedAt").value(containsString(date.format(formatter))));
+
+    }
+
+    @Test
+    @DisplayName("When POST with ToDos, Then ToDos")
+    public void whenPostingToDos_ThenSuccessfulAndReturnsToDos() throws Exception {
+
+        createToDos();
+        createTasks();
+        toDo1.setTasks(tasks);
+
+        Gson gson = new Gson();

```

Comment:
Can gson be not used in actual controllers as well, replacing Jackson

Reply:
Done